### PR TITLE
FT.AGGREGATE initial support ( GROUPBY REDUCE | SORTBY | APPLY | FILTER ).+ clauses and FT.CURSOR initial support

### PR DESCRIPTION
**Pull request by filipecosta90**
Hi there, starting this PR so that we can revise the Aggregation API. 
Note: Please disregard for now the tests of Aggregate since they were built with the ftsb queries in mind. I'll add the examples of [ RediSearch test_aggregate.py](https://github.com/RediSearch/RediSearch/blob/master/src/pytest/test_aggregate.py ) while you're revising the function signatures, etc... 

Regarding the changes on this WIP PR:
- [add] Adding aggregate.go and reducer.go. The main addition comes from aggregate.go with support for ( GROUPBY and REDUCE | SORTBY | APPLY | FILTER ).+ clauses
- [add] Extended client.go with the Aggregate method, that receives an AggregateQuery and returns the number of aggregate results, the Array Response and error if any. Each element of the Array response is an array of string's.
- [add] Extended SortingKey inside query.go with the Constructor and the Serialize methods. Both Query and AggregateQuery make usage of it. Adjusted `func (q Query) serialize() redis.Args `accordingly.
 - [add] Extended Paging inside query.go with the Constructor and the Serialize methods. Both Query and AggregateQuery make usage of it. Adjusted `func (q Query) serialize() redis.Args `accordingly.

### CODE REVIEW

File: redisearch/aggregate.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+// GroupBy
+type GroupBy struct {
+	Fields   []string
+	Reducers []Reducer
+	Paging   *Paging
+}
+
+func NewGroupBy(field string) *GroupBy {
```
Comment (MeirShpilraien) [inline redisearch/aggregate.go]: Why not creating an empty GROUPBY and then expose a function to add fields?

File: redisearch/reducer.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+// NewReducer creates a new Reducer object
+func NewReducerAlias(name GroupByReducers, args []string, alias string ) *Reducer {
+	return &Reducer{
+		Name:  name,
+		Alias: alias,
+		Args:  args,
+	}
+}
```
Comment (MeirShpilraien) [inline redisearch/reducer.go]: I think its not needed because we have the SetAlias function

File: redisearch/aggregate.go (hunk 1 above)
Comment (filipecosta90) [reply redisearch/aggregate.go]: done. please revise the new commit. Do you like AddFields in the following manner? It prevents us from having to create two nearly equal methods. 
```
func (g *GroupBy) AddFields(fields interface{}) *GroupBy
```

**Comment by filipecosta90** [comment]
@MeirShpilraien after your illustration on the total Aggregate results error, in-placed the changes to correct it. Please review.

**Comment by filipecosta90** [comment]
@MeirShpilraien, the latest commit:
- removes the FTSB queries from this test suite ( to make it simpler and independent ).
- refactors the processing of FT.AGGREGATE response since we might have room for improvement on that part in the future. 
- includes a more unsafe but faster aggregate response processor `go test -v -run=XXX -bench=. ./...` should get you more info about it. For now, I would keep the safer ProcessAggregateResponse.

**Comment by filipecosta90** [comment]
@MeirShpilraien and @mnunberg added support for FT.CURSOR in this PR as well.

**Pull request merged**